File: backend/Ledgerly.Application/DTOs/Customer/CustomerDto.cs
namespace Ledgerly.Application.DTOs.Customer;

public class CustomerDto
{
    public Guid Id { get; set; }
    public Guid OrganizationId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Contact { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: app/lib/models/customer/customer_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'customer_dto.g.dart';

@JsonSerializable()
class CustomerDto {
  final String id;
  final String organizationId;
  final String name;
  final String? contact;
  final String? notes;
  @JsonKey(name: 'createdAt')
  final DateTime createdAt;

  CustomerDto({
    required this.id,
    required this.organizationId,
    required this.name,
    this.contact,
    this.notes,
    required this.createdAt,
  });

  factory CustomerDto.fromJson(Map<String, dynamic> json) =>
      _$CustomerDtoFromJson(json);

  Map<String, dynamic> toJson() => _$CustomerDtoToJson(this);
}

